Client/Frontend/Home/HistoryPanel.swift

```diff
@@ -103,6 +102,7 @@ class HistoryPanel: UIViewController, HomePanel {
             self.syncedTabsButton.descriptionLabel.hidden = false
             self.syncedTabsButton.updateConstraints()
         }
+        return succeed()
```

Comment:
Instead of returning the `Deferred` from `self.profile.getCachedClientsAndTabs()`, you're returning a deferred with nothing in it. What's happening is the upon block above that is chained after this method isn't actually called after the `getCachedClientsAndTabs` call but is called immediately.

You'll want to return a `Deferred` resulting from the completion of the `self.profile.getCachedClientsAndTabs().uponQueue`. You can do something like:

```swift
return self.profile.getCachedClientsAndTabs().bindQueue(dispatch_get_main_queue()) {
    ...
    return succeed()
}
```
`bind` is similar to `upon` except that it returns a `Deferred` at the end instead of a value.

However, I would probably look to clean this further by consolidating the UI changes into a single method instead of having it split across this one and the upon block in `viewDidLoad`. One way is to slim down the `updateNumberOfSyncedDevices` call to only fetch the count, then have another method just for UI:

```swift 
func fetchSyncedDevicesCount() -> Deferred<Maybe<Int>> {
    return chainDeferred(self.profile.getCachedClientsAndTabs()) { deferMaybe($0.count) }
}
```

This way you can eliminate some of the duplication you have in updating the state of the `syncedTabsButton`.